src/components/navigation/section-document-or-browser.tsx

```diff
@@ -172,9 +184,95 @@ export const SectionDocumentOrBrowser: React.FC<IProps> = observer(function Sect
       tabsExtraClassNames={{"chat-open": isChatOpen}}
       onSelect={handleTabSelect}
       selectedIndex={subTabIndex}
-      renderSubTabPanel={subTab =>
-        renderDocumentView(subTab) || renderDocumentBrowserView(subTab)
-      }
+      renderSubTabPanel={subTab => renderDocumentView(subTab) || renderDocumentBrowserView(subTab)}
     />
   );
 });
+
+interface DocumentBrowserScrollerProps {
+  subTab: ISubTabSpec;
+  tabSpec: NavTabModelType;
+  openDocumentKey: string;
+  onSelectDocument: (document: DocumentModelType) => void;
+}
+const DocumentBrowserScroller =
+    ({subTab, tabSpec, openDocumentKey, onSelectDocument}: DocumentBrowserScrollerProps) => {
+  const [scrollerCollapsed, setScrollerCollapsed] = useState(false);
+  const [panelWidth, setPanelWidth] = useState(0);
+  const [scrollWidth, setScrollWidth] = useState(0);
+  const [scrollLeft, setScrollLeft] = useState(0);
+  const documentScrollerRef = useRef<HTMLDivElement>(null);
+  const [scrollToLocation, setScrollToLocation] = useState(0);
+
+  useEffect(()=>{
+    const scrollerEl = documentScrollerRef.current;
+    if (scrollerEl) {
+      setPanelWidth(scrollerEl.getBoundingClientRect().width);
+    }
+  },[documentScrollerRef]);
+
+  const handleScrollTo = (side: string) => {
+    const direction = side ==="left" ? -1 : 1;
+    setScrollToLocation((prevState) => {
+      const tempScrollTo = prevState + (direction * panelWidth);
+      if (tempScrollTo === 0) {
+        return (direction * panelWidth);
+      } else {
+        return tempScrollTo;
+      }
+    });
+  };
+
+  const handleCollapseScroller = () => {
+    setScrollerCollapsed(!scrollerCollapsed);
+  };
+
+  return (
+    <>
+      <div className={classNames("scroller", {"collapsed": scrollerCollapsed})} ref={documentScrollerRef}>
+        {(scrollLeft > 0) &&
+            <ScrollEndControl side={"left"} collapsed={scrollerCollapsed} tab={tabSpec.tab}
+                onScroll={handleScrollTo} />
+        }
+        <DocumentCollectionList
+            subTab={subTab}
+            tabSpec={tabSpec}
+            horizontal={true}
+            collapsed={scrollerCollapsed}
+            selectedDocument={openDocumentKey}
+            scrollToLocation={scrollToLocation}
+            onSelectDocument={onSelectDocument}
+            setScrollWidth={setScrollWidth}
+            setScrollLeft={setScrollLeft}
+        />
+        {((scrollWidth - scrollLeft - panelWidth) > 0) &&
```

Comment:
I think this makes the intention more clear here. Does that make sense to you?
```suggestion
        {scrollLeft < scrollWidth - panelWidth &&
```